File: src/Ccmc.Core/Services/ProjectSearch.cs
using Ccmc.Core.Models;

namespace Ccmc.Core.Services;

/// <summary>Search-box matching: case-insensitive substring on name or description.</summary>
public static class ProjectSearch
{
    public static bool Matches(ProjectInfo project, string term) =>
        project.Name.Contains(term, StringComparison.OrdinalIgnoreCase) ||
        project.Description.Contains(term, StringComparison.OrdinalIgnoreCase);
}

File: electron/src/core/projects/projectSearch.ts
import type { ProjectInfo } from '../models'

/** Search-box matching: case-insensitive substring on name or description. */
export function projectMatches(project: ProjectInfo, term: string): boolean {
  return (
    project.name.toLowerCase().includes(term.toLowerCase()) ||
    project.description.toLowerCase().includes(term.toLowerCase())
  )
}
